From: Adam Conrad <adconrad@ubuntu.com>
Date: Wed, 25 May 2022 18:22:00 +0800
Subject: Tidy up linux-headers installation and duplication

ubuntu-chroot_headers_tidy.patch: Mark all the linux-headers-$(ABI)
packages as autoinstalled, and then autoremove cruft (LP: #1029556)
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1029556
Forwarded: no
Last-Update: 2012-08-20
---
 scripts/build/lb_chroot | 6 ++++++
 scripts/build/lb_clean  | 1 +
 2 files changed, 7 insertions(+)

diff --git a/scripts/build/lb_chroot b/scripts/build/lb_chroot
index ada67c3..4144166 100755
--- a/scripts/build/lb_chroot
+++ b/scripts/build/lb_chroot
@@ -62,6 +62,12 @@ do
 
 	if [ "${_PASS}" = install ]
 	then
+		Chroot chroot "dpkg -l linux-headers-3* linux-headers-4*" 2>/dev/null \
+			| awk '/^i/ {print $2}' > chroot.headers
+		for i in $(cat chroot.headers); do
+			Chroot chroot "apt-mark auto $i"
+		done
+		Chroot chroot "apt-get --purge -y autoremove"
 		Chroot chroot "dpkg-query -W" > chroot.packages.install
 	fi
 done
diff --git a/scripts/build/lb_clean b/scripts/build/lb_clean
index 4b7c23b..e500a8c 100755
--- a/scripts/build/lb_clean
+++ b/scripts/build/lb_clean
@@ -88,6 +88,7 @@ do
 			${LB_ROOT_COMMAND} rm -rf chroot chroot.tmp
 
 			rm -f chroot.packages.live chroot.packages.install
+			rm -f chroot.headers
 
 			rm -f .build/chroot*
 			;;
